File: radarrsonarrm3usync/model/Configuration.py
from pydantic import BaseModel
from typing import Optional
from pathlib import Path


class Configuration(BaseModel):
    """Complete application configuration model."""
    radarr_enabled: bool = False
    radarr_base_url: str = ""
    radarr_api_key: str = ""
    
    sonarr_enabled: bool = False
    sonarr_base_url: str = ""
    sonarr_api_key: str = ""
    

File: src/types/Configuration.ts
/**
 * Complete application configuration model
 */
export default interface Configuration {
  radarr_enabled: boolean;
  radarr_base_url: string;
  radarr_api_key: string;
  radarr_base_dir: string;
  sonarr_enabled: boolean;
  sonarr_base_url: string;
  sonarr_api_key: string;
  sonarr_base_dir: string;
}
